### PR TITLE
CORE-1276: Add public key recovery from digest + signature

### PR DESCRIPTION
**Pull request by bgoring**

### CODE REVIEW

File: WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c [hunk 1]
```diff
@@ -105,6 +105,36 @@ wkWalletConnectorSignDataETH (
...
+    BRKey k;
+
+    *status = WK_WALLET_CONNECTOR_STATUS_OK;
+    if (sizeof(UInt256) != digestLength) {
+        *status = WK_WALLET_CONNECTOR_STATUS_INVALID_DIGEST;
+        return NULL;
+    }
+    if (digestLength != 65) {
+        *status = WK_WALLET_CONNECTOR_STATUS_INVALID_SIGNATURE;
+    }
```
Comment (EBGToo) [inline WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c:127]: Should be `signatureLength`

File: WalletKitSwift/WalletKit/WKWalletManager.swift [hunk 2]
```diff
@@ -638,7 +639,28 @@ public final class WalletConnector {
     public func recover (digest: Digest, signature: Signature) -> Result<Key, WalletConnectorError> {
         guard core == digest.core, core == signature.core else { return Result.failure(.unknownEntity) }
 
-        return Result.failure(.unrecoverableKey)
+        return digest.withUnsafeBytes { (digestBytes: UnsafeRawBufferPointer) -> Result<Key, WalletConnectorError> in
+            let digestAddr = digestBytes.baseAddress?.assumingMemoryBound(to:UInt8.self)
+            let digestLength = digestBytes.count
```
Comment (EBGToo) [inline WalletKitSwift/WalletKit/WKWalletManager.swift:644]: This doesn't compile?

It doesn't follow the `withUnsafeBytes` pattern.  Should be, maybe:
```
return digest.data32.withUnsafeBytes ...
```

File: WalletKitSwift/WalletKit/WKWalletManager.swift [hunk 3]
```diff
@@ -638,7 +639,28 @@ public final class WalletConnector {
...
+            
+            return signature.withUnsafeBytes { (signatureBytes: UnsafeRawBufferPointer) -> Result<Key, WalletConnectorError> in
+                let signatureAddr = signatureBytes.baseAddress?.assumingMemoryBound(to:UInt8.self);
+                let signatureLength = signatureBytes.length
```
Comment (EBGToo) [inline WalletKitSwift/WalletKit/WKWalletManager.swift:648]: Doesn't compile?

Comment (EBGToo) [reply WalletKitSwift/WalletKit/WKWalletManager.swift:648]: And `length` is not a property/function in Swift.  Should be `signatureBytes.count`.

File: WalletKitSwift/WalletKit/WKWalletManager.swift [hunk 4]
```diff
@@ -638,7 +639,28 @@ public final class WalletConnector {
...
+                                                      digestLength,
+                                                      signatureAddr,
+                                                      signatureLength,
+                                                      &status)
+                if (key == nil) {
+                    return Result.failure(WalletConnectorError(core: status))
+                }
+                
+                // Return key will own the walletkit native key memory hereafter
+                return Result.success(Key(core: key));
```
Comment (EBGToo) [inline WalletKitSwift/WalletKit/WKWalletManager.swift:661]: `key` is `uint8_t*` but `Key(core: key)` expects `BRKey`?

File: WalletKitCore/include/WKWalletConnector.h [hunk 5]
```diff
@@ -104,6 +107,26 @@ wkWalletConnectorSignData (
...
+ */
+extern uint8_t*
+wkWalletConnectorRecoverKey (
+        WKWalletConnector       connector,
+        const uint8_t           *digest,
+        size_t                  digestLength,
+        const uint8_t           *signature,
+        size_t                  signatureLength,
+        WKWalletConnectorStatus *status         );
+
```
Comment (EBGToo) [inline WalletKitCore/include/WKWalletConnector.h:129]: There is no implementation of this function?

**Review by EBGToo — CHANGES_REQUESTED** [review]

File: WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c (hunk 1 above)
Comment (bgoring) [reply WalletKitCore/src/walletkit/handlers/eth/WKWalletConnectorETH.c:127]: fixed

File: WalletKitCore/include/WKWalletConnector.h (hunk 5 above)
Comment (bgoring) [reply WalletKitCore/include/WKWalletConnector.h:129]: It not likely to work without that! Added.

File: WalletKitSwift/WalletKit/WKWalletManager.swift (hunk 2 above)
Comment (bgoring) [reply WalletKitSwift/WalletKit/WKWalletManager.swift:644]: Missed these compile problems w/o WalletKitDemo scheme enabled. Fixed

File: WalletKitSwift/WalletKit/WKWalletManager.swift (hunk 3 above)
Comment (bgoring) [reply WalletKitSwift/WalletKit/WKWalletManager.swift:648]: Fixed

File: WalletKitSwift/WalletKit/WKWalletManager.swift (hunk 4 above)
Comment (bgoring) [reply WalletKitSwift/WalletKit/WKWalletManager.swift:661]: Hold on. I have the interface wrong. It should return a WKKey. Fixed

**Review by EBGToo — APPROVED** [review]

**Pull request merged**